Concuerror 0.16 ('99e189d') started at 27 Sep 2016 16:13:08.
 Options:
  [{after_timeout,infinity},
   {assertions_only,false},
   {assume_racing,false},
   {depth_bound,500},
   {disable_sleep_sets,false},
   {dpor,optimal},
   {entry_point,{readers_2,test,[]}},
   {files,["/home/stavros/git/Concuerror/tests/suites/bounding_tests/src/readers_2.erl"]},
   {ignore_error,[]},
   {instant_delivery,true},
   {interleaving_bound,infinity},
   {keep_going,true},
   {non_racing_system,[]},
   {print_depth,20},
   {scheduling,round_robin},
   {scheduling_bound,1},
   {scheduling_bound_type,delay},
   {show_races,false},
   {strict_scheduling,false},
   {symbolic_names,true},
   {timeout,infinity},
   {treat_as_normal,[]}]
################################################################################
Interleaving #1
--------------------------------------------------------------------------------
Errors found:
* Blocked at a 'receive' (when all other processes have exited):
    P in readers_2.erl line 14
--------------------------------------------------------------------------------
Event trace:
   1: P: tab = ets:new(tab, [public,named_table])
    in readers_2.erl line 9
   2: P: P.1 = erlang:spawn(erlang, apply, [#Fun<readers_2.'-readers/1-fun-2-'.0>,[]])
    in erlang.erl line 2603
   3: P: P.2 = erlang:spawn(erlang, apply, [#Fun<readers_2.'-readers/1-fun-2-'.0>,[]])
    in erlang.erl line 2603
   4: P: P.3 = erlang:spawn(erlang, apply, [#Fun<readers_2.'-readers/1-fun-2-'.0>,[]])
    in erlang.erl line 2603
   5: P: P.4 = erlang:spawn(erlang, apply, [#Fun<readers_2.'-readers/1-fun-2-'.0>,[]])
    in erlang.erl line 2603
   6: P: P.5 = erlang:spawn(erlang, apply, [#Fun<readers_2.'-readers/1-fun-2-'.0>,[]])
    in erlang.erl line 2603
   7: P: P.6 = erlang:spawn(erlang, apply, [#Fun<readers_2.'-readers/1-fun-2-'.0>,[]])
    in erlang.erl line 2603
   8: P: P.7 = erlang:spawn(erlang, apply, [#Fun<readers_2.'-readers/1-fun-0-'.0>,[]])
    in erlang.erl line 2603
   9: P.1: [] = ets:lookup(tab, x)
    in readers_2.erl line 11
  10: P.1: exits normally
  11: P.2: [] = ets:lookup(tab, x)
    in readers_2.erl line 11
  12: P.2: exits normally
  13: P.3: [] = ets:lookup(tab, x)
    in readers_2.erl line 11
  14: P.3: exits normally
  15: P.4: [] = ets:lookup(tab, x)
    in readers_2.erl line 11
  16: P.4: exits normally
  17: P.5: [] = ets:lookup(tab, x)
    in readers_2.erl line 11
  18: P.5: exits normally
  19: P.6: [] = ets:lookup(tab, x)
    in readers_2.erl line 11
  20: P.6: exits normally
  21: P.7: true = ets:insert(tab, {x,42})
    in readers_2.erl line 10
  22: P.7: exits normally
################################################################################
Interleaving #2
--------------------------------------------------------------------------------
Errors found:
* Blocked at a 'receive' (when all other processes have exited):
    P in readers_2.erl line 14
--------------------------------------------------------------------------------
Event trace:
   1: P: tab = ets:new(tab, [public,named_table])
    in readers_2.erl line 9
   2: P: P.1 = erlang:spawn(erlang, apply, [#Fun<readers_2.'-readers/1-fun-2-'.0>,[]])
    in erlang.erl line 2603
   3: P: P.2 = erlang:spawn(erlang, apply, [#Fun<readers_2.'-readers/1-fun-2-'.0>,[]])
    in erlang.erl line 2603
   4: P: P.3 = erlang:spawn(erlang, apply, [#Fun<readers_2.'-readers/1-fun-2-'.0>,[]])
    in erlang.erl line 2603
   5: P: P.4 = erlang:spawn(erlang, apply, [#Fun<readers_2.'-readers/1-fun-2-'.0>,[]])
    in erlang.erl line 2603
   6: P: P.5 = erlang:spawn(erlang, apply, [#Fun<readers_2.'-readers/1-fun-2-'.0>,[]])
    in erlang.erl line 2603
   7: P: P.6 = erlang:spawn(erlang, apply, [#Fun<readers_2.'-readers/1-fun-2-'.0>,[]])
    in erlang.erl line 2603
   8: P: P.7 = erlang:spawn(erlang, apply, [#Fun<readers_2.'-readers/1-fun-0-'.0>,[]])
    in erlang.erl line 2603
   9: P.1: [] = ets:lookup(tab, x)
    in readers_2.erl line 11
  10: P.1: exits normally
  11: P.2: [] = ets:lookup(tab, x)
    in readers_2.erl line 11
  12: P.2: exits normally
  13: P.3: [] = ets:lookup(tab, x)
    in readers_2.erl line 11
  14: P.3: exits normally
  15: P.4: [] = ets:lookup(tab, x)
    in readers_2.erl line 11
  16: P.4: exits normally
  17: P.5: [] = ets:lookup(tab, x)
    in readers_2.erl line 11
  18: P.5: exits normally
  19: P.7: true = ets:insert(tab, {x,42})
    in readers_2.erl line 10
  20: P.7: exits normally
  21: P.6: [{x,42}] = ets:lookup(tab, x)
    in readers_2.erl line 11
  22: P.6: exits normally
################################################################################
Interleaving #3
--------------------------------------------------------------------------------
Errors found:
* Blocked at a 'receive' (when all other processes have exited):
    P in readers_2.erl line 14
--------------------------------------------------------------------------------
Event trace:
   1: P: tab = ets:new(tab, [public,named_table])
    in readers_2.erl line 9
   2: P: P.1 = erlang:spawn(erlang, apply, [#Fun<readers_2.'-readers/1-fun-2-'.0>,[]])
    in erlang.erl line 2603
   3: P: P.2 = erlang:spawn(erlang, apply, [#Fun<readers_2.'-readers/1-fun-2-'.0>,[]])
    in erlang.erl line 2603
   4: P: P.3 = erlang:spawn(erlang, apply, [#Fun<readers_2.'-readers/1-fun-2-'.0>,[]])
    in erlang.erl line 2603
   5: P: P.4 = erlang:spawn(erlang, apply, [#Fun<readers_2.'-readers/1-fun-2-'.0>,[]])
    in erlang.erl line 2603
   6: P: P.5 = erlang:spawn(erlang, apply, [#Fun<readers_2.'-readers/1-fun-2-'.0>,[]])
    in erlang.erl line 2603
   7: P: P.6 = erlang:spawn(erlang, apply, [#Fun<readers_2.'-readers/1-fun-2-'.0>,[]])
    in erlang.erl line 2603
   8: P: P.7 = erlang:spawn(erlang, apply, [#Fun<readers_2.'-readers/1-fun-0-'.0>,[]])
    in erlang.erl line 2603
   9: P.1: [] = ets:lookup(tab, x)
    in readers_2.erl line 11
  10: P.1: exits normally
  11: P.2: [] = ets:lookup(tab, x)
    in readers_2.erl line 11
  12: P.2: exits normally
  13: P.3: [] = ets:lookup(tab, x)
    in readers_2.erl line 11
  14: P.3: exits normally
  15: P.4: [] = ets:lookup(tab, x)
    in readers_2.erl line 11
  16: P.4: exits normally
  17: P.6: [] = ets:lookup(tab, x)
    in readers_2.erl line 11
  18: P.6: exits normally
  19: P.7: true = ets:insert(tab, {x,42})
    in readers_2.erl line 10
  20: P.7: exits normally
  21: P.5: [{x,42}] = ets:lookup(tab, x)
    in readers_2.erl line 11
  22: P.5: exits normally
################################################################################
Interleaving #4
--------------------------------------------------------------------------------
Errors found:
* Blocked at a 'receive' (when all other processes have exited):
    P in readers_2.erl line 14
--------------------------------------------------------------------------------
Event trace:
   1: P: tab = ets:new(tab, [public,named_table])
    in readers_2.erl line 9
   2: P: P.1 = erlang:spawn(erlang, apply, [#Fun<readers_2.'-readers/1-fun-2-'.0>,[]])
    in erlang.erl line 2603
   3: P: P.2 = erlang:spawn(erlang, apply, [#Fun<readers_2.'-readers/1-fun-2-'.0>,[]])
    in erlang.erl line 2603
   4: P: P.3 = erlang:spawn(erlang, apply, [#Fun<readers_2.'-readers/1-fun-2-'.0>,[]])
    in erlang.erl line 2603
   5: P: P.4 = erlang:spawn(erlang, apply, [#Fun<readers_2.'-readers/1-fun-2-'.0>,[]])
    in erlang.erl line 2603
   6: P: P.5 = erlang:spawn(erlang, apply, [#Fun<readers_2.'-readers/1-fun-2-'.0>,[]])
    in erlang.erl line 2603
   7: P: P.6 = erlang:spawn(erlang, apply, [#Fun<readers_2.'-readers/1-fun-2-'.0>,[]])
    in erlang.erl line 2603
   8: P: P.7 = erlang:spawn(erlang, apply, [#Fun<readers_2.'-readers/1-fun-0-'.0>,[]])
    in erlang.erl line 2603
   9: P.1: [] = ets:lookup(tab, x)
    in readers_2.erl line 11
  10: P.1: exits normally
  11: P.2: [] = ets:lookup(tab, x)
    in readers_2.erl line 11
  12: P.2: exits normally
  13: P.3: [] = ets:lookup(tab, x)
    in readers_2.erl line 11
  14: P.3: exits normally
  15: P.5: [] = ets:lookup(tab, x)
    in readers_2.erl line 11
  16: P.5: exits normally
  17: P.6: [] = ets:lookup(tab, x)
    in readers_2.erl line 11
  18: P.6: exits normally
  19: P.7: true = ets:insert(tab, {x,42})
    in readers_2.erl line 10
  20: P.7: exits normally
  21: P.4: [{x,42}] = ets:lookup(tab, x)
    in readers_2.erl line 11
  22: P.4: exits normally
################################################################################
Interleaving #5
--------------------------------------------------------------------------------
Errors found:
* Blocked at a 'receive' (when all other processes have exited):
    P in readers_2.erl line 14
--------------------------------------------------------------------------------
Event trace:
   1: P: tab = ets:new(tab, [public,named_table])
    in readers_2.erl line 9
   2: P: P.1 = erlang:spawn(erlang, apply, [#Fun<readers_2.'-readers/1-fun-2-'.0>,[]])
    in erlang.erl line 2603
   3: P: P.2 = erlang:spawn(erlang, apply, [#Fun<readers_2.'-readers/1-fun-2-'.0>,[]])
    in erlang.erl line 2603
   4: P: P.3 = erlang:spawn(erlang, apply, [#Fun<readers_2.'-readers/1-fun-2-'.0>,[]])
    in erlang.erl line 2603
   5: P: P.4 = erlang:spawn(erlang, apply, [#Fun<readers_2.'-readers/1-fun-2-'.0>,[]])
    in erlang.erl line 2603
   6: P: P.5 = erlang:spawn(erlang, apply, [#Fun<readers_2.'-readers/1-fun-2-'.0>,[]])
    in erlang.erl line 2603
   7: P: P.6 = erlang:spawn(erlang, apply, [#Fun<readers_2.'-readers/1-fun-2-'.0>,[]])
    in erlang.erl line 2603
   8: P: P.7 = erlang:spawn(erlang, apply, [#Fun<readers_2.'-readers/1-fun-0-'.0>,[]])
    in erlang.erl line 2603
   9: P.1: [] = ets:lookup(tab, x)
    in readers_2.erl line 11
  10: P.1: exits normally
  11: P.2: [] = ets:lookup(tab, x)
    in readers_2.erl line 11
  12: P.2: exits normally
  13: P.4: [] = ets:lookup(tab, x)
    in readers_2.erl line 11
  14: P.4: exits normally
  15: P.5: [] = ets:lookup(tab, x)
    in readers_2.erl line 11
  16: P.5: exits normally
  17: P.6: [] = ets:lookup(tab, x)
    in readers_2.erl line 11
  18: P.6: exits normally
  19: P.7: true = ets:insert(tab, {x,42})
    in readers_2.erl line 10
  20: P.7: exits normally
  21: P.3: [{x,42}] = ets:lookup(tab, x)
    in readers_2.erl line 11
  22: P.3: exits normally
################################################################################
Interleaving #6
--------------------------------------------------------------------------------
Errors found:
* Blocked at a 'receive' (when all other processes have exited):
    P in readers_2.erl line 14
--------------------------------------------------------------------------------
Event trace:
   1: P: tab = ets:new(tab, [public,named_table])
    in readers_2.erl line 9
   2: P: P.1 = erlang:spawn(erlang, apply, [#Fun<readers_2.'-readers/1-fun-2-'.0>,[]])
    in erlang.erl line 2603
   3: P: P.2 = erlang:spawn(erlang, apply, [#Fun<readers_2.'-readers/1-fun-2-'.0>,[]])
    in erlang.erl line 2603
   4: P: P.3 = erlang:spawn(erlang, apply, [#Fun<readers_2.'-readers/1-fun-2-'.0>,[]])
    in erlang.erl line 2603
   5: P: P.4 = erlang:spawn(erlang, apply, [#Fun<readers_2.'-readers/1-fun-2-'.0>,[]])
    in erlang.erl line 2603
   6: P: P.5 = erlang:spawn(erlang, apply, [#Fun<readers_2.'-readers/1-fun-2-'.0>,[]])
    in erlang.erl line 2603
   7: P: P.6 = erlang:spawn(erlang, apply, [#Fun<readers_2.'-readers/1-fun-2-'.0>,[]])
    in erlang.erl line 2603
   8: P: P.7 = erlang:spawn(erlang, apply, [#Fun<readers_2.'-readers/1-fun-0-'.0>,[]])
    in erlang.erl line 2603
   9: P.1: [] = ets:lookup(tab, x)
    in readers_2.erl line 11
  10: P.1: exits normally
  11: P.3: [] = ets:lookup(tab, x)
    in readers_2.erl line 11
  12: P.3: exits normally
  13: P.4: [] = ets:lookup(tab, x)
    in readers_2.erl line 11
  14: P.4: exits normally
  15: P.5: [] = ets:lookup(tab, x)
    in readers_2.erl line 11
  16: P.5: exits normally
  17: P.6: [] = ets:lookup(tab, x)
    in readers_2.erl line 11
  18: P.6: exits normally
  19: P.7: true = ets:insert(tab, {x,42})
    in readers_2.erl line 10
  20: P.7: exits normally
  21: P.2: [{x,42}] = ets:lookup(tab, x)
    in readers_2.erl line 11
  22: P.2: exits normally
################################################################################
Interleaving #7
--------------------------------------------------------------------------------
Errors found:
* Blocked at a 'receive' (when all other processes have exited):
    P in readers_2.erl line 14
--------------------------------------------------------------------------------
Event trace:
   1: P: tab = ets:new(tab, [public,named_table])
    in readers_2.erl line 9
   2: P: P.1 = erlang:spawn(erlang, apply, [#Fun<readers_2.'-readers/1-fun-2-'.0>,[]])
    in erlang.erl line 2603
   3: P: P.2 = erlang:spawn(erlang, apply, [#Fun<readers_2.'-readers/1-fun-2-'.0>,[]])
    in erlang.erl line 2603
   4: P: P.3 = erlang:spawn(erlang, apply, [#Fun<readers_2.'-readers/1-fun-2-'.0>,[]])
    in erlang.erl line 2603
   5: P: P.4 = erlang:spawn(erlang, apply, [#Fun<readers_2.'-readers/1-fun-2-'.0>,[]])
    in erlang.erl line 2603
   6: P: P.5 = erlang:spawn(erlang, apply, [#Fun<readers_2.'-readers/1-fun-2-'.0>,[]])
    in erlang.erl line 2603
   7: P: P.6 = erlang:spawn(erlang, apply, [#Fun<readers_2.'-readers/1-fun-2-'.0>,[]])
    in erlang.erl line 2603
   8: P: P.7 = erlang:spawn(erlang, apply, [#Fun<readers_2.'-readers/1-fun-0-'.0>,[]])
    in erlang.erl line 2603
   9: P.2: [] = ets:lookup(tab, x)
    in readers_2.erl line 11
  10: P.2: exits normally
  11: P.3: [] = ets:lookup(tab, x)
    in readers_2.erl line 11
  12: P.3: exits normally
  13: P.4: [] = ets:lookup(tab, x)
    in readers_2.erl line 11
  14: P.4: exits normally
  15: P.5: [] = ets:lookup(tab, x)
    in readers_2.erl line 11
  16: P.5: exits normally
  17: P.6: [] = ets:lookup(tab, x)
    in readers_2.erl line 11
  18: P.6: exits normally
  19: P.7: true = ets:insert(tab, {x,42})
    in readers_2.erl line 10
  20: P.7: exits normally
  21: P.1: [{x,42}] = ets:lookup(tab, x)
    in readers_2.erl line 11
  22: P.1: exits normally
################################################################################
Exploration completed!
################################################################################
Warnings:
--------------------------------------------------------------------------------
Some interleavings were not considered due to schedule bounding.

################################################################################
Info:
--------------------------------------------------------------------------------
Instrumented readers_2
Instrumented io_lib
Instrumented gen_server
Instrumented lists
Instrumented erlang
You can see pairs of racing instructions (in the report and --graph) with '--show_races true'
Continuing after error (-k)

################################################################################
Done at 27 Sep 2016 16:13:09 (Exit status: error)
  Summary: 7 errors, 7/7 interleavings explored (the scheduling bound was reached)
